pkg/metrics/envoy.go

```diff
@@ -20,7 +20,7 @@ var envoyQueries = map[string]string{
 		rate(
 			envoy_cluster_upstream_rq{
 				kubernetes_namespace="{{ .Namespace }}",
-				kubernetes_pod_name=~"{{ .Name }}-[0-9a-zA-Z]+(-[0-9a-zA-Z]+)"
+				envoy_cluster_name=~"{{ .Name }}-canary"
```

Comment:
This breaks App Mesh, the envoy_cluster_name doesn't have this format. Please create a crossover metrics observer.

Reply:
I've discussed with stefan about this in Slack and the fix has been pushed.